standards/SPIFFE_Broker_API.md

```diff
@@ -0,0 +1,411 @@
+
+The `JWTSVIDResponse` response message consists of a mandatory `svids` field, which MUST contain one or more `JWTSVID` messages.
+
+All fields in the `JWTSVID` message are mandatory, with the exception of the `hint` field. When the `hint` field is set (i.e. non-empty), SPIFFE Broker API servers MUST ensure its value is unique amongst the set of returned SVIDs in any given `JWTSVIDResponse` message. In the event that a SPIFFE Broker API client encounters more than one `JWTSVID` message with the same `hint` value set, then the first message in the list SHOULD be selected.
+
+If the referenced workload does not exist or is not entitled to receive any JWT-SVIDs, then the server SHOULD respond with the "FailedPrecondition" gRPC status code. See [Section 7](#7-workload-reference-error) for details. Under such a case, the Broker MAY attempt to reconnect with another call to the `FetchJWTSVID` RPC after a backoff.
+
+#### 6.2.2 FetchJWTBundles
+
+The `FetchJWTBundles` RPC streams back JWT bundles for the workload to the Broker. These bundles MUST only be used to authenticate JWT-SVIDs and MUST only be used for operations involving the referenced workload. They MUST NOT be used for any other workload.
+
+The `JWTBundlesRequest` request message contains a reference to the workload.
+
+The `JWTBundlesResponse` response message consists of a mandatory `bundles` field, which MUST contain at least the JWT bundle for the trust domain in which the server resides.
+
+The returned bundles are encoded as a standard JWK Set as defined by [RFC 7517](https://tools.ietf.org/html/rfc7517) containing the JWT-SVID signing keys for the trust domain. These keys may only represent a subset of the keys present in the SPIFFE trust bundle for the trust domain. The server MUST NOT include keys with other uses in the returned JWT bundles.
+
+If the referenced workload does not exist or is not entitled to receive any JWT bundles, then the server SHOULD respond with the "FailedPrecondition" gRPC status code. See [Section 7](#7-workload-reference-error) for details. Under such a case, the Broker MAY attempt to reconnect with another call to the `FetchJWTBundles` RPC after a backoff.
+
+As mentioned in [Stream Responses](#43-stream-responses), each `JWTBundlesResponse` response contains the complete set of authorized JWT bundles of the workload at that point in time. As such, if the server redacts bundles from a subsequent response that was in context of the referenced workload the Broker SHOULD cease using the redacted bundles. This includes situations where the server returns a Permission denied, where the Broker is expected to drop all previous received bundles.
+
+# 7. Workload Reference Error
```

Comment:
Do you have any examples of what this would look like in Go (or similar) on the implementation? I'm usually familiar with `google.rpc.Status` being used to convey additional error details for gRPC and it's not too clear how this fits in with that or replaces it.